macOS/DuckDuckGo/NetworkProtection/AppTargets/BothAppTargets/NetworkProtectionNavBarButtonModel.swift

```diff
@@ -133,16 +133,22 @@ final class NetworkProtectionNavBarButtonModel: NSObject, ObservableObject {
 
     @MainActor
     func updateVisibility() {
-        guard !isPinned,
-              !popoverManager.isShown,
-              !isHavingConnectivityIssues else {
+        Task { @MainActor in
+            guard let canStartVPN = try? await vpnGatekeeper.canStartVPN() else {
+                // If there's an error, don't make any changes
```

Comment:
Just remain in the previous state in this scenario.  Shortcut pins are persisted so no need to change anything here.